tests/syntax/test_requirements.py

```diff
@@ -315,3 +315,42 @@ def test_static_intersection_violation_disabled():
         ego = new Object at 0@0
         new Object at 1@0, with allowCollisions True
     """)
+
+## Tests for random properties that have special effects on constructing requirements
+def test_random_allowCollisions():
+    for _ in range(5):
+        sampleSceneFrom("""
+            new Object with allowCollisions Uniform(True, False)
+            new Object with allowCollisions Uniform(True, False)
+        """, maxIterations=10)
+
+@pytest.mark.slow
+def test_random_occlusion():
+    for _ in range(5):
```

Comment:
As above, I'm not sure that just repeating the test 5 times is enough. It would probably also be good to add an assertion that in all returned scenes the wall has `occluding` false.